test/Concurrency/actor_isolation.swift

```diff
@@ -1257,6 +1263,32 @@ func testGlobalActorInheritance() {
   acceptAsyncSendableClosureInheriting {
     counter += 1 // ok
   }
+
+  acceptAsyncSendableClosureInheritingAlways {
+    counter += 1 // ok
+  }
+}
+
+@available(SwiftStdlib 5.1, *)
+func testIsolatedParameter1(_: isolated any Actor, v: inout Int) {
+  acceptAsyncSendableClosureInheriting {
+    v += 1 // expected-warning {{mutable capture of 'inout' parameter 'v' is not allowed in concurrently-executing code}}
+  }
+
+  acceptAsyncSendableClosureInheritingAlways {
+    v += 1 // Ok
+  }
+}
+
+@available(SwiftStdlib 5.1, *)
+func testIsolatedParameter2(_: isolated (any Actor)? = #isolation, v: inout Int) {
+  acceptAsyncSendableClosureInheriting {
+    v += 1 // expected-warning {{mutable capture of 'inout' parameter 'v' is not allowed in concurrently-executing code}}
+  }
+
+  acceptAsyncSendableClosureInheritingAlways {
+    v += 1 // Ok
+  }
 }
```

Comment:
Minor followup: Would it be worth adding a test for `@MainActor` as well, it'll "just work" because `_inheritActorContext` itself should be doing that, but in order make sure we don't accidentally regress that in `_inheritActorContext(always)` somehow?

Reply:
Ah the diff had hidden the `testGlobalActorInheritance` case, we got that covered then 👍